src/main/java/net/fabricmc/loom/configuration/providers/mappings/tiny/TinyMappingsSpec.java

```diff
@@ -0,0 +1,35 @@
+/*
+ * This file is part of fabric-loom, licensed under the MIT License (MIT).
+ *
+ * Copyright (c) 2021 FabricMC
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+package net.fabricmc.loom.configuration.providers.mappings.tiny;
+
+import net.fabricmc.loom.configuration.providers.mappings.MappingContext;
+import net.fabricmc.loom.configuration.providers.mappings.MappingsSpec;
+
+public record TinyMappingsSpec(String dependencyNotation) implements MappingsSpec<TinyMappingLayer> {
```

Comment:
Im not a fan of this being a string, imo it should be some sort of file resolver that can handle maven deps and local files. With local files it's hashcode should be calulated using the file's hash so it will regen when the local file changes.

I did start looking into this but got a bit burnt out for what ever reason: https://github.com/FabricMC/fabric-loom/compare/dev/0.9...modmuss50:mappingiolayer

Reply:
I wanted to use `Dependency`, but resolving that would tie the mapping context to a Gradle `Project` (so no mocking?)

Also, would it be possible to use a truncated sha256 hash of the resulting mapping file? That feels more reliable than java hash codes.